activesupport/lib/active_support/cache/file_store.rb

```diff
@@ -78,23 +77,21 @@ def read_entry(key, options)
         end
 
         def write_entry(key, entry, options)
-          file_name = key_file_path(key)
-          return false if options[:unless_exist] && File.exist?(file_name)
-          ensure_cache_path(File.dirname(file_name))
-          File.atomic_write(file_name, cache_path) {|f| Marshal.dump(entry, f)}
+          return false if options[:unless_exist] && File.exist?(key)
+          ensure_cache_path(File.dirname(key))
+          File.atomic_write(key, cache_path) {|f| Marshal.dump(entry, f)}
```

Comment:
An example of the compat break: an overwritten `write_entry` wouldn't resolve the key to a `file_name`.